src/processors/operation-processor.ts

```diff
@@ -36,11 +36,54 @@ export default class OperationProcessor<ResourceT extends Resource> {
 
     if (result !== undefined) {
       eagerLoadedData = await this.eagerLoad(op, result);
+      eagerLoadedData = await this.computeRelationshipProperties(op, eagerLoadedData);
     }
 
     return this.convertToResources(op, result, eagerLoadedData);
   }
 
+  async computeRelationshipProperties(op, eagerLoadedData) {
```

Comment:
I'm thinking we should probably break down this function into smaller pieces. Take a look at the following comments.